sdk/rust/oak/src/lib.rs

```diff
@@ -452,12 +453,27 @@ pub fn run_command_loop<N: CommandHandler, R: Iterator<Item = N::Command>>(
 /// Trait implemented by structs that correspond to entrypoint of Wasm nodes.
 ///
 /// This statically ties together the Wasm entrypoint name (i.e. the name of the Wasm exported
-/// function) to the associated constant of this trait.
+/// function) and the type of messages that the node accepts with this trait.
 ///
 /// This trait is automatically implemented for types used as part of [`entrypoint_command_handler`]
 /// entrypoint definitions.
 pub trait WasmEntrypoint {
+    /// Name of the WebAssembly exported function that corresponds to the entrypoint of this Node.
     const ENTRYPOINT_NAME: &'static str;
+
+    /// Type of message that the Node accepts on its inbound channel.
+    type Message: Encodable + Decodable;
+}
+
+/// Trait implemented by structs that require an initialization step by receiving data over the
+/// inbound channel (usually wrapped within a [`oak_io::InitWrapper`] message).
+pub trait WithInit {
```

Comment:
Will devs need to use this trait by themselves or it will just be automatically used in existing macros?

Reply:
The macros use it, but nothing stops devs to use it manually if they want to, it's effectively just a constructor. In general I don't think it's very useful to manually create a struct which is normally part of a node anyways.